compiler/lib/src/main/scala/analysis/CheckSemantics/CheckSpecLocs.scala

```diff
@@ -18,19 +18,44 @@ object CheckSpecLocs
   override def defArrayAnnotatedNode(a: Analysis, aNode: Ast.Annotated[AstNode[Ast.DefArray]]) = {
     val (_, node, _) = aNode
     val name = node.data.name
-    checkSpecLoc(a, Ast.SpecLoc.Type, name, node)
+    val qualName = a.getQualifiedName(Symbol.Array(aNode))
+    val id = aNode._2.id
+    for {
+      a <- checkSpecLoc(a, Ast.SpecLoc.Type, name, node)
+      a <- checkDictionarySpecLoc(a, Ast.SpecLoc.Type, qualName, id, node.data.isDictionaryDef)
+    } yield a
+  }
+
+  override def defAliasTypeAnnotatedNode(a: Analysis, aNode: Ast.Annotated[AstNode[Ast.DefAliasType]]) = {
+    val (_, node, _) = aNode
+    val name = node.data.name
+    val qualName = a.getQualifiedName(Symbol.AliasType(aNode))
+    val id = aNode._2.id
+    for {
+      a <- checkDictionarySpecLoc(a, Ast.SpecLoc.Type, qualName, id, node.data.isDictionaryDef)
+    } yield a
```

Comment:
Alias types weren't initially covered in this pass. Do we need to run location specifier check for aliases? I only included the dictionary location specifier check.